server/rest/src/main/java/org/infinispan/rest/resources/ContainerResource.java

```diff
@@ -533,13 +541,27 @@ private CompletionStage<RestResponse> shutdown(RestRequest restRequest) {
    }
 
    private CompletionStage<RestResponse> listenConfig(RestRequest request) {
+      return streamConfigurationAndEvents(request, false);
+   }
+
+   private CompletionStage<RestResponse> listenLifecycle(RestRequest request) {
+      return streamConfigurationAndEvents(request, true);
+   }
+
+   private CompletionStage<RestResponse> streamConfigurationAndEvents(RestRequest request, boolean includeLifecycle) {
       MediaType mediaType = negotiateMediaType(request, APPLICATION_YAML, APPLICATION_JSON, APPLICATION_XML);
       boolean includeCurrentState = Boolean.parseBoolean(request.getParameter("includeCurrentState"));
       EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
       NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
       ConfigurationListener listener = new ConfigurationListener(cacheManager, mediaType, includeCurrentState);
       responseBuilder.contentType(TEXT_EVENT_STREAM).entity(listener.getEventStream());
-      return cacheManager.addListenerAsync(listener).thenApply(v -> responseBuilder.build());
+
+      CompletionStage<Void> cs = cacheManager.addListenerAsync(listener);
```

Comment:
I think we need to do this (plus subsequent calls) via a `SecurityAction` for when Authorization is enabled, as `addListenerAsync` checks for the LISTEN permission.

The original container SSE endpoint was missing similar checks, see https://github.com/infinispan/infinispan/pull/10232 for details.

Reply:
Updated, thanks! I've created a new one for adding the listener into the logger and kind of cloned the existent one to use the async method.